### PR TITLE
Doubleclick Fast Fetch gct parameter

### PR DESCRIPTION
**Pull request by keithwrightbos**
For Doubleclick Fast Fetch, populate gct parameter used as part of DFP UI preview.

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -636,10 +639,16 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+              PAGE_LEVEL_PARAMS_,
+              {
+                'gct': this.getLocationQueryParameterValue('google_preview') ||
+                    null,
+              });
           return googleAdUrl(
               this, DOUBLECLICK_BASE_URL, startTime, Object.assign(
                   this.getBlockParameters_(), rtcParams,
-                  this.buildIdentityParams_(), PAGE_LEVEL_PARAMS_));
+                  this.buildIdentityParams_(), pageParams));
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: does this not need to be done for SRA?

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 2]
```diff
@@ -153,6 +153,9 @@ let sraRequests = null;
     }} */
 let TroubleshootData;
 
+/** @private {?JsonObject} */
+let windowLocationQueryParameters = null;
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: nit: don't think you need to instantiate to null here, you could just do let windowLocationQueryParameters.

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 1 above)
Comment (keithwrightbos) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Good catch!  I make a getPageParameters_ function to ensure consistency while allowing for easy testing.

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 2 above)
Comment (keithwrightbos) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 3]
```diff
@@ -535,6 +538,13 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {resolver, rejector, promise};
   }
 
+  /** @return {!Object<string,string|boolean|number>} */
+  getPageParameters_() {
+    return Object.assign(PAGE_LEVEL_PARAMS_, {
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: can get rid of PAGE_LEVEL_PARAMS_ entirely, and move everything in here

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 4]
```diff
@@ -535,6 +538,13 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {resolver, rejector, promise};
   }
 
+  /** @return {!Object<string,string|boolean|number>} */
+  getPageParameters_() {
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Not totally crazy about the naming of this, as we already have googlePageParameters and pageLevelParameters. Not sure what would be better names though...

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js [hunk 5]
```diff
@@ -574,6 +574,18 @@ describes.realWin('amp-ad-network-doubleclick-impl', realWinConfig, env => {
         });
       });
     });
+    it('should have google_preview parameter', () => {
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js]: can we add an sra test in test-doubleclick-sra too? just a simple addition to an existing test that that the gct param gets added

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 3 above)
Comment (keithwrightbos) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Done

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js (hunk 4 above)
Comment (keithwrightbos) [reply extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Yeah I hear you... googlePageParameters are those shared by AdSense and GPT.  I renamed pageLevelParameters to googPageLevelParameters

File: extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js (hunk 5 above)
Comment (keithwrightbos) [reply extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js]: Done

**Pull request merged**